groovy/gradle.java/src/org/netbeans/modules/gradle/java/api/GradleJavaProject.java

```diff
@@ -0,0 +1,150 @@
+    public static final String CLASSIFIER_SOURCES = "sources"; //NOI18N
+
+    File mainJar;
+    Set<File> testClassesRoots;
+    Set<File> coverageData;
+    Map<String, File> archives;
+
+    transient Map<File, GradleJavaSourceSet> fileToSourceSetCache;
+
+    Map<String, GradleJavaSourceSet> sourceSets = new LinkedHashMap<>();
+
+    protected GradleJavaProject() {
+    }
+
+    public File getMainJar() {
+        return mainJar;
+    }
+
+    public File getArchive(String classifier) {
+        return archives != null ? archives.get(classifier) : null;
+    }
+
+    public Set<File> getCoverageData() {
+        return coverageData;
+    }
+
+    public Set<File> getTestClassesRoots() {
+        return testClassesRoots;
+    }
+
+    public Map<String, GradleJavaSourceSet> getSourceSets() {
+        return Collections.unmodifiableMap(sourceSets);
+    }
+
+    public GradleJavaSourceSet getMainSourceSet() {
+        return sourceSets.get(GradleJavaSourceSet.MAIN_SOURCESET_NAME);
+    }
+    
+    public GradleJavaSourceSet containingSourceSet(File f) {
+        if (fileToSourceSetCache == null) {
+            fileToSourceSetCache = new WeakHashMap<>();
+        }
+        GradleJavaSourceSet ret = fileToSourceSetCache.get(f);
+        if (ret == null) {
+            for (GradleJavaSourceSet sourceSet : sourceSets.values()) {
+                if (sourceSet.contains(f)) {
+                    ret = sourceSet;
+                    fileToSourceSetCache.put(f, ret);
+                    break;
+                }
+            }
+        }
+        return ret;
+    }
+
+    protected GradleJavaSourceSet createSourceSet(String name) {
+        GradleJavaSourceSet ret = new GradleJavaSourceSet(name);
+        sourceSets.put(name, ret);
+        return ret;
+    }
+
+    void resolveSourceSetDependencies() {
+        for (GradleJavaSourceSet source : sourceSets.values()) {
+            if (source.compileClassPath != null) {
+                for (File cpEntry : source.getCompileClassPath()) {
+                    for (GradleJavaSourceSet depends : sourceSets.values()) {
+                        if (depends == source) {
+                            continue;
+                        }
+                        if (depends.outputContains(cpEntry)) {
+                            if (source.sourceDependencies.isEmpty()) {
+                                source.sourceDependencies = new LinkedHashSet<>();
+                            }
+                            source.sourceDependencies.add(depends);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public static GradleJavaProject get(Project project) {
+        GradleJavaProject ret = null;
+        NbGradleProject gprj = NbGradleProject.get(project);
+        if (gprj != null) {
+            ret = get(gprj);
+        }
+        return ret;
+    }
+
+    public static GradleJavaProject get(NbGradleProject project) {
```

Comment:
I would keep just the `get(Project)` method. At least right now, before reading the rest of the diff.

Reply:
Good advice! Thanks!